lib/src/pubspec.dart

```diff
@@ -116,9 +116,30 @@ class Pubspec {
     }
   }
 
-  factory Pubspec.fromJson(Map json) => _$PubspecFromJson(json);
+  factory Pubspec.fromJson(Map json, {bool lenient = false}) {
```

Comment:
Isn't `permissive` a better way to write it?

Reply:
Dart SDK doesn't have either, and Java's date parsing uses lenient.

Reply:
Also shouldn't this be the default mode?

Reply:
I think defaulting to strict mode is better.